From 8e9cfbf602af71a1dd056fdffaa072e88af69ba6 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Thu, 12 Sep 2024 21:33:30 +0300
Subject: [PATCH] PASN: Fix pasn-resp fuzzing tester build

The recently added calls to src/ap/pmksa_cache_auth.c needs to be faked
to allow pasn-resp to be built without having to pull in multiple
additional files from src/ap.

Fixes: b7de417c8a47 ("PASN: Define PMKSA helper functions for initiator and responder")
Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 tests/fuzzing/pasn-resp/pasn-resp.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/tests/fuzzing/pasn-resp/pasn-resp.c b/tests/fuzzing/pasn-resp/pasn-resp.c
index 8d44de66a..26100542a 100644
--- a/tests/fuzzing/pasn-resp/pasn-resp.c
+++ b/tests/fuzzing/pasn-resp/pasn-resp.c
@@ -41,6 +41,30 @@ pmksa_cache_auth_get(struct rsn_pmksa_cache *pmksa,
 }
 
 
+struct rsn_pmksa_cache *
+pmksa_cache_auth_init(void (*free_cb)(struct rsn_pmksa_cache_entry *entry,
+				      void *ctx), void *ctx)
+{
+	return NULL;
+}
+
+
+void pmksa_cache_auth_deinit(struct rsn_pmksa_cache *pmksa)
+{
+}
+
+
+void pmksa_cache_free_entry(struct rsn_pmksa_cache *pmksa,
+			    struct rsn_pmksa_cache_entry *entry)
+{
+}
+
+
+void pmksa_cache_auth_flush(struct rsn_pmksa_cache *pmksa)
+{
+}
+
+
 static int pasn_send_mgmt(void *ctx, const u8 *data, size_t data_len,
 			  int noack, unsigned int freq, unsigned int wait)
 {
-- 
2.43.0

